src/__tests__/gilded_rose_spec.js

```diff
@@ -1,11 +1,114 @@
-import { Shop, Item } from '../gilded_rose';
+import { Shop, Item } from "../gilded_rose";
 
-describe("Gilded Rose", function () {
+describe("Gilded Rose", function() {
+  describe("Item", () => {
+    it("initiates correctly", () => {
+      const item = new Item("foo", 10, 40);
+      expect(item).toEqual(new Item("foo", 10, 40));
+    });
+  });
 
-    it("should foo", function () {
-        const gildedRose = new Shop([new Item("foo", 0, 0)]);
-        const items = gildedRose.updateQuality();
-        expect(items[0].name).toEqual("fixme");
+  describe("Shop", () => {
+    it("stores given items", () => {
+      const item = new Item("foo", 10, 40);
+      const shopItems = [item, item];
+      const shop = new Shop(shopItems);
+      expect(shop.items).toEqual(shopItems);
     });
 
+    describe("updateQuality", () => {
+      it("lowers sellIn and quality on each run", () => {
+        const items = [new Item("foo", 10, 40), new Item("bar", 4, 20)];
+        const shop = new Shop(items);
+        shop.updateQuality();
+
+        expect(shop.items).toEqual([
+          new Item("foo", 9, 39),
+          new Item("bar", 3, 19)
+        ]);
+      });
+
+      it("lowers quality by one if the sell date has not passed", () => {
+        const items = [new Item("foo", 2, 40)];
+        const shop = new Shop(items);
+        shop.updateQuality();
+        expect(shop.items).toEqual([new Item("foo", 1, 39)]);
+      });
+
+      it("lowers quality by one if the sell date has not passed", () => {
+        const items = [new Item("foo", 0, 40)];
+        const shop = new Shop(items);
+        shop.updateQuality();
+        expect(shop.items).toEqual([new Item("foo", -1, 38)]);
+      });
+
+      it("ensures quality of an item is never negative", () => {
```

Comment:
What is the quality in the setup of the test? There are two values. Both are 0. Which value is important.